docs/docs/kds-configuration.md

```diff
@@ -207,32 +158,94 @@ kds {
-code signing key must be trusted enclave clients because they have the ability
-to author a new enclave that is able to perform any action on data protected
-using the signing key.
-
-This example is probably most suitable where the service author is trusted, or
-when using an external auditor to validate enclave code and sign the enclave on
-the developer's behalf.
+#### Any enclave for a particular product signed using the same key
+In this permissive policy constraint example, access to the derived key is permitted for enclaves with a matching [product ID](enclave-configuration.md#productid) and signing key.
 
 ```groovy
 kds {
     persistenceKeySpec {
         policyConstraint {
-            constraint = "SEC:SECURE"
+            constraint = "SEC:STALE"
             useOwnCodeSignerAndProductID = true
         }
     }
 }
 ```
+
+This constraint has fewer operational implications, as the enclave code can be updated without losing access to persisted data.
+However, security is weakened with this constraint, as the enclave signer must be trusted not to sign an unsafe or malicious enclave.
+
+!!!Note
+    In this example, security may be enhanced by having an external auditor build and sign the enclave on the behalf of the developer.
+
+## KDS keys for Conclave Mail
+The KDS is also capable of providing keys for use with [Conclave Mail](mail.md) through use of a specially constructed [`PostOffice`](api/-conclave/com.r3.conclave.mail/-post-office/index.html).
+
+When using a KDS post office, the post office requests a public key from the KDS for use in a Conclave [`PostOffice`](api/-conclave/com.r3.conclave.mail/-post-office/index.html) and specifies which enclaves should be permitted access to the corresponding private key.
+Upon receiving Mail, the enclave requests the private key from the same KDS and uses it to decrypt the Mail object from the client.
+In this manner, users can create Mail objects which can be decrypted by any enclave that matches a given set of constraints, without first having to receive and attest to the [`EnclaveInstanceInfo`](api/-conclave/com.r3.conclave.common/-enclave-instance-info/index.html) of a specific enclave.
+
+### Configuring the Enclave
+To make use of KDS keys for Mail, the `kdsEnclaveConstraints` field in the enclaves `build.gradle` must be configured.
+This configuration alone is sufficient to use KDS keys with Mail.
+
+```groovy
+...
+conclave {
+    kds {
+        kdsEnclaveConstraints = "<KDS enclave policy constraints>"
+    } 
+}
+...
+```
+
+!!!Note
+    For the R3 KDS, the following constraint should be used: `S:B4CDF6F4FA5B484FCA82292CE340FF305AA294F19382178BEA759E30E7DCFE2D PROD:1 SEC:STALE`.
+
+### Defining a KDS key spec
+To make use of KDS keys with Mail, the client must construct an appropriately configured [`PostOffice`](api/-conclave/com.r3.conclave.mail/-post-office/index.html) object with which to encrypt Mail.
```

Comment:
Why this duplication of the phrase "To make use of KDS keys with Mail"?

Reply:
Yeah, it's a bit meh. Removed.